js/ui/placesManager.js

```diff
@@ -106,29 +108,101 @@ PlaceDeviceInfo.prototype = {
-                notification.setTransient(true);
-
+            Main.messageTray.add(source);
+            let notification = new MessageTray.Notification(source, msg1, msg2);
+            notification.setTransient(true);
+            if (withButton) {
                 notification.addButton('system-undo', _("Retry"));
                 notification.connect('action-invoked', Lang.bind(this, this.remove));
-                source.notify(notification);
+            }
+            source.notify(notification);
+        } else {
+            if (msg2)
+                global.log(msg1 + ': ' + msg2);
+            else
+                global.log(msg1);
+        }
+    },
+
+    _stopFinish: function(drive, res) {
+        if (DEBUG) global.log("PlacesManager: **_stopFinish**");
+        let driveName = drive.get_name();  // Ex: USB Flash Drive
+        let unixDevice = drive.get_identifier('unix-device'); // Ex: /dev/sdc
+        let msg1 = _("%s (%s) has just been stopped.").format(driveName, this.name);
+        let msg2 = _("Device %s can be turned off, if necessary.").format(unixDevice);
+        try {
+            drive.stop_finish(res);
+            this._sendNotification(msg1, msg2);
+        } catch(e) {
+            this._sendNotification(_("Unable to stop the drive %s (%s)").format(drive.get_name(), this.name), e.message, true);
+        }
+    },
+
+    _ejectFinish: function(source, res, is_drive) {
+        if (DEBUG) global.log("PlacesManager: **_ejectFinish**");
+        let msg1;
+        let msg2 = null;
+
+        if (is_drive) {
+            let driveName = source.get_name();  // Ex: USB Flash Drive
+            let unixDevice = source.get_identifier('unix-device'); // Ex: /dev/sdc
+            msg1 = _("%s (%s) can be safely unplugged.").format(driveName, this.name);
+            msg2 = _("Device %s can be removed.").format(unixDevice);
+        } else {
+            msg1 = _("%s (%s) has just been ejected.").format(source.get_name(), this.name);
+        }
+        try {
+            source.eject_with_operation_finish(res);
+            this._sendNotification(msg1, msg2);
+        } catch(e) {
+            this._sendNotification(_("Unable to eject the drive %s (%s)").format(source.get_name(), this.name), e.message, true);
+        }
+    },
+
+    _removeFinish: function(o, res, data) {
+        if (DEBUG) global.log("PlacesManager: **_removeFinish**");
+        let msg1 = "";
+        let msg2 = null;
+
+        if (this._mount.can_eject()) {
+            msg1 = _("%s (%s) can be safely unplugged").format(o.get_name(), this.name);
+            msg2 = _("Device can be removed");
+            try {
+                this._mount.eject_with_operation_finish(res);
+                this._sendNotification(msg1, msg2);
+            } catch(e) {
+                this._sendNotification(_("Failed to eject %s (%s)").format(o.get_name(), this.name), e.message, true);
+            }
```

Comment:
Is this code ever reached?  You call `_ejectFinish() if `this._mount.can_eject()` in the `remove()` method, so I don't think this would ever be reached.  I may be mistaken (mounts, drives, volumes are confusing).

Reply:
for some reason it doesn't display the block of code i wanted - this is what I am referring to:
https://github.com/linuxmint/cinnamon/pull/8988/files/8b5cbb4b6c47fb32f1f7dac013597aec99da29e1#diff-65655e0b47c0d12eb7b07fb200e3d23dR190-R198